pennylane/_qubit_device.py

```diff
@@ -304,10 +304,10 @@ def execute(self, circuit, **kwargs):
                 and getattr(self, "_CPP_BINARY_AVAILABLE", False)
             )
             diagonalizing_gates = self._get_diagonalizing_gates(circuit) if is_lightning else None
-            if is_lightning and diagonalizing_gates:
+            if is_lightning and diagonalizing_gates:  # pragma: no cover
```

Comment:
This should be left as is. These branches will be removed once all lightning devices are using the new API.